lib/plugins/helpers/network.rb

```diff
@@ -1,49 +1,76 @@
 # frozen_string_literal: true
 
-# Network Helpers for Plugin Class
+require 'socket'
+require 'timeout'
+
+# Network Helpers
 # Its main functions are to:
 #  1. Set the Bot IP Address based on Public and if in DIND
 #  2. Exposed Ports using user provided or dynamic port assignments
-class Plugin
-  def bot_ip
-    option = @config['managed'] ? 'private' : 'public' if @config['type'] == 'api'
-    option = 'private' unless @config['type'] == 'api'
+class Network
```

Comment:
I like this! Seems like a solid separate entity that is testable with minimal dependency injection of just the config hash.